p2p/peer.go

```diff
@@ -82,12 +83,13 @@ const (
 )
 
 // PeerEvent is an event emitted when peers are either added or dropped from
-// a p2p.Server
+// a p2p.Server or when a message is sent or received on a peer connection
 type PeerEvent struct {
```

Comment:
I think we will need more info of the message and not sure we necessarily care about the Code and size. 
What is we just optionally pass a `func(interface{}) ([]byte, bool) ` function to the eventer which would rlp-encode any 
data we wanna map the message to for different use cases.

Of course this is perfect for now as is.